src/acl.rs

```diff
@@ -0,0 +1,189 @@
+use std::fmt;
+use std::ops;
+
+use std::string::ToString;
+
+/// Describes the ability of a user to perform a certain action.
+///
+/// Permissions can be mixed together like integers with `|` and `&`.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[repr(u32)]
+pub enum Permission {
```

Comment:
Permissions are now an explicit object instead of being some integers in a `mod`.